packages/x-tree-view/src/TreeItem/TreeItem.types.ts

```diff
@@ -50,6 +50,12 @@ export interface TreeItemSlots extends TreeItemIconSlots {
    * @default TreeItemDragAndDropOverlay
    */
   dragAndDropOverlay?: React.ElementType;
+  /**
+   * The component that wraps the `iconContainer` slot when the item is in an error state.
+   * Warning: This slot is only useful when using the `<RichTreeView />` component.
```

Comment:
Is it ever useful without lazy loading?
If not we can probably improve the JSDoc to make that clear.